bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
```

Comment:
If that's the case why do we need to do *anything* with docs or roots right here? I'm trying not to duplicate what `_ModelInDocument` is doing, and it seems like we are at least partially doing that below.